FirebaseAppCheck/Sources/AppAttestProvider/Storage/FIRAppAttestArtifactStorage.h

```diff
@@ -24,16 +24,20 @@ NS_ASSUME_NONNULL_BEGIN
 /// Check token obtained with App Attest provider.
 @protocol FIRAppAttestArtifactStorageProtocol <NSObject>
 
-/// Set the artifact.
+/// Set the artifact. An artifact previously set for *any* key ID will replace the new one with the
+/// new key ID. The storage always stores as single artifact.
 /// @param artifact The artifact data to store. Pass `nil` to remove the stored artifact.
+/// @param keyID The App Attest key ID used to generate the artifact.
 /// @return An artifact that is resolved with the artifact data passed into the method in case of
 /// success or is rejected with an error.
-- (FBLPromise<NSData *> *)setArtifact:(nullable NSData *)artifact;
+- (FBLPromise<NSData *> *)setArtifact:(nullable NSData *)artifact forKey:(NSString *)keyID;
```

Comment:
Should the keyID have the `nonnull` attribute that is specified in the method's implementation? Does adding it here give any more type/nullability safety? ... I was wondering about this and unsure

Reply:
All variables placed between `NS_ASSUME_NONNULL_BEGIN` and `NS_ASSUME_NONNULL_END` are marked as `nonnull`, so we need to specify nullability explicitly only for nullable variables/parameters. Without `NS_ASSUME_NONNULL_...` we would have to add the nullability modifiers for each variable.

Reply:
Ohhh I see. I understand that (and the purpose of `NS_ASSUME_NONNULL_BEGIN`) much better now– thank you!